File: Python/extract_lists.py
import json

# Читання файлу spaces.json
with open('resources/spaces.json', 'r', encoding='utf-8') as file:
    data = json.load(file)

# Знаходження всіх об'єктів "lists" і виведення їх ID
for folder in data['folders']:
    if 'lists' in folder:
        for lst in folder['lists']:
            print(f"List ID: {lst['id']}")

# Виведення space.name та space.id для lists, де ім'я починається зі слова "test"
for folder in data['folders']:
    if 'lists' in folder:
        for lst in folder['lists']:
            if 'name' in lst and lst['name'].lower().startswith('test'):
                print(f"Space Name: {folder['space']['name']}")
                print(f"Space ID: {folder['space']['id']}")

File: JS/extract_lists.js
const fs = require('fs');

// Читання файлу spaces.json
fs.readFile('resources/spaces.json', 'utf8', (err, data) => {
    if (err) {
        console.error('Помилка читання файлу:', err);
        return;
    }

    try {
        // Парсинг JSON
        const spaces = JSON.parse(data);

        // Перевірка, чи є spaces масивом або об'єктом
        if (Array.isArray(spaces.folders)) {
            // Якщо folders - масив
            spaces.folders.forEach(space => {
                if (space.lists) {
                    space.lists.forEach(list => {
                        console.log('List ID:', list.id);
                    });
                }
            });

            spaces.folders.forEach(space => {
                if (space.lists) {
                    space.lists.forEach(list => {
                        if (list.name && list.name.toLowerCase().startsWith('test')) {
                            console.log('Space Name:', space.space.name);
                            console.log('Space ID:', space.space.id);
                        }
                    });
                }
            });
        } else {
            console.log('Не вдалося знайти масив folders у JSON');
        }

    } catch (parseErr) {
        console.error('Помилка парсингу JSON:', parseErr);
    }
});
